Tests/NIOSSLTests/TLSConfigurationTest.swift

```diff
@@ -650,92 +662,59 @@ class TLSConfigurationTest: XCTestCase {
         serverConfig.trustRoots = .file(tempFileDir) // Directory path.
         serverConfig.certificateVerification = .fullVerification
         
-        // Pulling in the logic for assertHandshakeSucceededEventLoop because there needs to be some custom handling
-        // due to the IP address and the potential for `CERTIFICATE_VERIFY_FAILED`.
-        var clientContext: NIOSSLContext!
         var serverContext: NIOSSLContext!
-        
-        XCTAssertNoThrow(clientContext = try NIOSSLContext(configuration: clientConfig))
         XCTAssertNoThrow(serverContext = try NIOSSLContext(configuration: serverConfig))
-
-        let group = MultiThreadedEventLoopGroup(numberOfThreads: 1)
-        defer {
-            XCTAssertNoThrow(try group.syncShutdownGracefully())
-        }
-
-        let eventHandler = ErrorCatcher<NIOSSLError>()
-        let handshakeHandler = HandshakeCompletedHandler()
-        let handshakeResultPromise = group.next().makePromise(of: Void.self)
-
-        let handshakeWatcher = WaitForHandshakeHandler(handshakeResultPromise: handshakeResultPromise)
-
-        var serverChannel: Channel!
-        var clientChannel: Channel!
-
-        XCTAssertNoThrow(serverChannel = try serverTLSChannel(context: serverContext, handlers: [], group: group))
- 
-        // Validation that the CA names are being sent here
-        // This is essentially the heart of this unit test.
+        // Only setup the serverContext here to define that our two certificate CA names were populated to the SSL_CTX.
         let countAfter = serverContext.getX509NameListCount()
         XCTAssertEqual(countAfter, 2, "CA Name List should be 2 after the Server Context is created")
         
-        XCTAssertNoThrow(clientChannel = try clientTLSChannel(context: clientContext, preHandlers:[], postHandlers: [eventHandler, handshakeWatcher, handshakeHandler], group: group, connectingTo: serverChannel.localAddress!))
+        // Delete all files that were created for this test.
+        XCTAssertNoThrow(try FileManager.default.removeItem(at: rootCAURLOne))
+        XCTAssertNoThrow(try FileManager.default.removeItem(at: rootCAURLTwo))
+        XCTAssertNoThrow(try FileManager.default.removeItem(at: URL(string: "file://" + rehashSymlinkNameOne)!))
+        XCTAssertNoThrow(try FileManager.default.removeItem(at: URL(string: "file://" + rehashSymlinkNameTwo)!))
```

Comment:
Can we aim to put these in a defer block so that we can be quite religious about cleaning them up?

Reply:
This is an **EXCELLENT** idea.

Reply:
This will prevent the need to clean up the directories by hand when something fails.  Thanks for calling that out.